profiles/services.py

```diff
@@ -290,7 +290,10 @@ def add_scores_from_data(user_stats: UserStats, score_data_list: list[ScoreData]
     (requires all dicts to have beatmap_id set along with usual score data)
     """
     # Remove potential duplicates from a top 100 play also being in the recent 50
-    unique_score_data_list = list(set(score_data_list))
+    unique_score_data_list = []
+    for score_data in score_data_list:
+        if score_data not in unique_score_data_list:
+            unique_score_data_list.append(score_data)
```

Comment:
Can't use `set()` to dedupe anymore since statistics (dict) isn't hashable.
Could get around it, but this is clear enough.